src/Dropdown/Dropdown.js

```diff
@@ -121,45 +120,42 @@ export default class Dropdown extends React.Component {
     }
   };
 
-  attachAutocloseListener() {
-    if (!this.props.autoclose) {
-      return;
-    }
-    document.addEventListener('click', this.autoclose);
-  }
-
-  detachAutocloseListener() {
-    document.removeEventListener('click', this.autoclose);
-  }
-
   autoclose = e => {
-    if (this.state.isOpen && !this.menuRef.current.contains(e.target)) {
+    if (this.state.isOpen) {
       this.hide();
     }
   };
 
+  handleBlur = () => {
+    this.timeoutId = setTimeout(this.autoclose, 500);
+  }
+
+  handleFocus = () => {
+    clearTimeout(this.timeoutId);
+  }
+
   render() {
     const { width, trigger, render, children } = this.props;
     const { isOpen } = this.state;
 
     const renderFn = render || children;
 
     return (
-      <Fragment>
-        <DropdownTrigger ref={this.triggerRef} aria-haspopup="true" aria-expanded={isOpen}>
+      <DropdownWrapper tabIndex={0} onBlur={this.handleBlur}>
+        <DropdownTrigger onClick={this.toggle} ref={this.triggerRef} aria-haspopup="true" aria-expanded={isOpen}>
           {trigger}
         </DropdownTrigger>
 
         <Portal>
           {isOpen && (
-            <DropdownMenu ref={this.menuRef} width={width}>
+            <DropdownMenu ref={this.menuRef} width={width} onFocus={this.handleFocus} tabIndex={0}>
```

Comment:
Since our dropdown menu is managed by popper.js, attaching the onFocus handler here helps to register focus events properly (it was not firing when attached to the DropdownWrapper).